### PR TITLE
CA-230: feat: implement AppShellPage with 4-tab layout, lazy loading, and project header app bar

### PR DESCRIPTION
**Pull request by ayanasamuel8**
### Summary
Introduces `AppShellPage` as the main app shell with 4-tab bottom navigation (Home, Calculations, Cost Estimation, Members). Implements lazy module loading and tab state preservation.

### Changes
- **app_shell_page.dart** - New shell page with:
  - Bottom navigation using `CoreBottomNavBar`
  - Lazy loading of tab modules via `TabModuleManager`
  - Tab state preservation using `Offstage` widgets
  - Project-aware app bar (switches based on current project)
  - Back button handling with tab-specific navigation
  - Stream subscription for reactive project ID updates

- **tab_module_manager.dart** - Simplified `CalculationsModule` and `MembersModule` by removing unnecessary `appBootstrap` dependency

- **app_en.arb** - Added localization keys: `appTitle`, `homeTab`, `calculationsTab`, `membersTab`

- **app_shell_page_test.dart** - Comprehensive widget tests covering:
  - Tab navigation and switching
  - Lazy loading verification
  - Tab state preservation (widget identity)
  - Project ID handling edge cases
  - App bar behavior
  - Project subscription reactivity

### Testing
- [x] Widget tests pass
- [x] Localization keys added

### CODE REVIEW

**Review by SimonDerejeW — CHANGES_REQUESTED** [review]

**Review by ripplearcgit — COMMENTED** [review]
**Upcoming integration point with `ProjectDropdownBloc` from Bazabizi's CA-113 chain**

Once Bazabizi's #232 (`ProjectDropdownBloc`) merges into `main`, `AppShellPage` can replace the hardcoded fallback project id (`'950e8400-e29b-41d4-a716-446655440001'` in `initState`) with a real project driven by the BLoC. The integration requires two small additions to `AppShellPage`:

1. Dispatch `ProjectDropdownStarted` in `initState` after `_moduleLoader.ensureTabModuleLoaded(ShellTab.home)` — this kicks off the real-time project stream from Supabase.

2. Wrap the existing `BlocBuilder<AppShellBloc, AppShellState>` with a `BlocListener<ProjectDropdownBloc, ProjectDropdownState>` that calls `_currentProjectNotifier.setCurrentProjectId(state.selectedProject?.id)` on `ProjectDropdownLoadSuccess`. This is the single wire between the dropdown and the persistent app bar.

No changes to `CurrentProjectNotifier`, `TabModuleManager`, or any other part of the shell are needed — the architecture is already set up correctly for this. The `ProjectDropdownBloc` is registered in `DashboardModule`, which is lazy-bound by `TabModuleManager` when the Home tab first loads, so it will be available in the DI container by the time the listener fires.

Nothing needs to change in this PR for now — flagging ahead of time so the integration is not a surprise when Bazabizi's chain lands.

**Review by ripplearcgit — CHANGES_REQUESTED** [review]
Simon's two prior `REQUEST_CHANGES` comments are both unresolved in the current commit. Addressing those plus three new blocking issues found in this review pass.

See inline comments for specifics.

File: lib/app/shell/app_shell_page.dart [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+import 'package:construculator/libraries/extensions/extensions.dart';
+import 'package:construculator/libraries/project/interfaces/current_project_notifier.dart';
+import 'package:construculator/libraries/project/presentation/project_ui_provider.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_modular/flutter_modular.dart';
+import 'package:ripplearc_coreui/ripplearc_coreui.dart';
+
+class AppShellPage extends StatefulWidget {
```
Comment (ripplearcgit) [inline lib/app/shell/app_shell_page.dart]: Doc please

File: lib/app/shell/app_shell_page.dart [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    _bloc.add(AppShellTabSelected(index));
+  }
+
+  Widget _buildTabRoot(ShellTab tab) {
+    if (!_moduleLoader.isLoaded(tab)) {
+      return const Center(child: CircularProgressIndicator());
+    }
+    switch (tab) {
+      case ShellTab.home:
+        return const DashboardPage();
```
[the remark this answers is not in the document]
Comment (ripplearcgit) [reply lib/app/shell/app_shell_page.dart]: If we do keep it, the circular must come from the CoreUI. please double check with @SimonDerejeW.

**Review by ripplearcgit — CHANGES_REQUESTED** [review]

**Review by ripplearcgit — CHANGES_REQUESTED** [review]
Changes Requested

The architectural fixes in this round are solid — `TabModuleProvider` restored, estimation bloc wiring delegated to `EstimationModule.landingPage`, `EstimationCoreModule` / `EstimationRoutesModule` split clean. Three blockers remain before merge.

**Review by ripplearcgit — COMMENTED** [review]
🟡 Non-blocking doc comments — three new public classes are missing `///`.

File: lib/app/shell/app_shell_page.dart [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+class AppShellPage extends StatefulWidget {
+  const AppShellPage({super.key});
+
+  @override
+  State<AppShellPage> createState() => _AppShellPageState();
+}
+
+class _AppShellPageState extends State<AppShellPage> {
+  final AppShellBloc _bloc = AppShellBloc();
```
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply lib/app/shell/app_shell_page.dart]: :+1:

File: lib/app/shell/app_shell_page.dart [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+      case ShellTab.home:
+        return const DashboardPage();
+      case ShellTab.calculations:
+        return const CalculationsPage();
+      case ShellTab.estimation:
+        final projectId = _projectId;
+        if (projectId == null || projectId.isEmpty) {
+          return const SizedBox.shrink();
+        }
+        return MultiBlocProvider(
```
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply lib/app/shell/app_shell_page.dart]: :+1:

File: lib/app/shell/app_shell_page.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply lib/app/shell/app_shell_page.dart]: :+1:

File: lib/app/shell/app_shell_page.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply lib/app/shell/app_shell_page.dart]: :+1:

File: lib/app/shell/tab_module_manager.dart [hunk 5]
```diff
@@ -1,37 +1,33 @@
...
-
-  static Map<ShellTab, TabModuleProvider> _defaultProviders() => {
-    ShellTab.home: const NoOpTabModuleProvider(),
-    ShellTab.calculations: const NoOpTabModuleProvider(),
-    ShellTab.estimation: const NoOpTabModuleProvider(),
-    ShellTab.members: const NoOpTabModuleProvider(),
-  };
+  TabModuleManager(this.appBootstrap);
 
   Future<void> ensureTabModuleLoaded(ShellTab tab) async {
```
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply lib/app/shell/tab_module_manager.dart:34]: :+1:

File: lib/app/shell/app_shell_page.dart [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+      projectId: projectId,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<AppShellBloc, AppShellState>(
+      bloc: _bloc,
+      builder: (context, state) {
+        return PopScope(
```
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply lib/app/shell/app_shell_page.dart:119]: :+1:

File: lib/app/shell/app_shell_page.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply lib/app/shell/app_shell_page.dart:69]: :+1:

File: test/app/shell/app_shell_page_test.dart [hunk 7]
```diff
@@ -0,0 +1,353 @@
...
+    testWidgets('bottom navigation bar is always visible', (tester) async {
+      await tester.pumpWidget(makeApp());
+      await tester.pumpAndSettle();
+
+      expect(find.byType(CoreBottomNavBar), findsOneWidget);
+
+      for (final tabLabel in ['Calculations', 'Members']) {
+        await tapTabByLabel(tester, tabLabel);
+        expect(find.byType(CoreBottomNavBar), findsOneWidget);
+      }
```
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply test/app/shell/app_shell_page_test.dart:181]: :+1:

File: lib/app/shell/app_shell_page.dart (hunk 1 above)
Comment (ayanasamuel8) [reply lib/app/shell/app_shell_page.dart:21]: :+1:

File: lib/app/shell/app_shell_page.dart [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+    (_) => GlobalKey<NavigatorState>(),
+  );
+
+  StreamSubscription<String?>? _projectSubscription;
+  String? _projectId;
+
+  @override
+  void initState() {
+    super.initState();
+    _projectId = _currentProjectNotifier.currentProjectId;
```
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply lib/app/shell/app_shell_page.dart]: :+1:

File: lib/app/shell/app_shell_page.dart [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+
+class AppShellPage extends StatefulWidget {
+  const AppShellPage({super.key});
+
+  @override
+  State<AppShellPage> createState() => _AppShellPageState();
+}
+
+class _AppShellPageState extends State<AppShellPage> {
+  final AppShellBloc _bloc = AppShellBloc();
```
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply lib/app/shell/app_shell_page.dart]: :+1:

File: lib/app/shell/app_shell_page.dart [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+  final CurrentProjectNotifier _currentProjectNotifier =
+      Modular.get<CurrentProjectNotifier>();
+  final TabModuleManager _moduleLoader = Modular.get<TabModuleManager>();
```
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply lib/app/shell/app_shell_page.dart]: :+1:

File: lib/app/shell/app_shell_page.dart [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+import 'package:construculator/libraries/project/interfaces/current_project_notifier.dart';
+import 'package:construculator/libraries/project/presentation/project_ui_provider.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_modular/flutter_modular.dart';
+import 'package:ripplearc_coreui/ripplearc_coreui.dart';
+
+class AppShellPage extends StatefulWidget {
+  const AppShellPage({super.key});
```
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply lib/app/shell/app_shell_page.dart:22]: :+1:

File: lib/app/shell/app_shell_page.dart (hunk 8 above)
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply lib/app/shell/app_shell_page.dart]: :+1:

File: lib/app/shell/shell_module.dart [hunk 12]
```diff
@@ -1,30 +1,46 @@
...
+import 'package:construculator/app/shell/app_shell_page.dart';
+import 'package:construculator/app/shell/module_model.dart';
 import 'package:construculator/app/shell/tab_module_manager.dart';
 import 'package:construculator/features/estimation/estimation_module.dart';
+import 'package:construculator/features/estimation/estimation_routes_module.dart';
 import 'package:construculator/libraries/router/routes/estimation_routes.dart';
-import 'package:flutter/material.dart';
 import 'package:flutter_modular/flutter_modular.dart';
 
+class EstimationTabModuleProvider implements TabModuleProvider {
```
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply lib/app/shell/shell_module.dart]: :+1:

File: lib/features/estimation/estimation_routes_module.dart [hunk 13]
```diff
@@ -0,0 +1,38 @@
+import 'package:construculator/app/app_bootstrap.dart';
+import 'package:construculator/features/auth/auth_module.dart';
+import 'package:construculator/features/estimation/presentation/pages/cost_estimation_details_page.dart';
+import 'package:construculator/libraries/estimation/estimation_core_module.dart';
+import 'package:construculator/libraries/router/guards/auth_guard.dart';
+import 'package:construculator/libraries/router/routes/estimation_routes.dart';
+import 'package:flutter_modular/flutter_modular.dart';
+
```
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply lib/features/estimation/estimation_routes_module.dart:8]: :+1:

File: lib/libraries/estimation/estimation_core_module.dart [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+import 'package:construculator/features/estimation/data/repositories/cost_estimation_repository_impl.dart';
+import 'package:construculator/features/estimation/domain/repositories/cost_estimation_log_repository.dart';
+import 'package:construculator/features/estimation/domain/repositories/cost_estimation_repository.dart';
+import 'package:construculator/features/estimation/domain/usecases/add_cost_estimation_usecase.dart';
+import 'package:construculator/libraries/auth/auth_library_module.dart';
+import 'package:construculator/libraries/time/clock_module.dart';
+import 'package:flutter_modular/flutter_modular.dart';
+
+class EstimationCoreModule extends Module {
+  final AppBootstrap appBootstrap;
```
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply lib/libraries/estimation/estimation_core_module.dart]: :+1:

File: lib/features/estimation/presentation/pages/cost_estimation_landing_page.dart [hunk 15]
```diff
@@ -44,6 +48,24 @@ class _CostEstimationLandingPageState extends State<CostEstimationLandingPage> {
     super.initState();
     _scrollController = ScrollController()..addListener(_onScroll);
     _router = Modular.get<AppRouter>();
+    _projectNotifier = Modular.get<CurrentProjectNotifier>();
```
Comment (SimonDerejeW) [inline lib/features/estimation/presentation/pages/cost_estimation_landing_page.dart]: This is too much logic on the UI.

File: lib/features/estimation/presentation/pages/cost_estimation_landing_page.dart [hunk 16]
```diff
@@ -15,18 +17,17 @@ import 'package:construculator/libraries/estimation/domain/estimation_error_type
...
 import 'package:ripplearc_coreui/ripplearc_coreui.dart';
 
 class CostEstimationLandingPage extends StatefulWidget {
-  final String projectId;
-
   static const double _buttonBottomRatio = 0.135;
   static const double _buttonRightRatio = 0.05;
 
-  const CostEstimationLandingPage({super.key, required this.projectId});
+  const CostEstimationLandingPage({super.key});
```
Comment (SimonDerejeW) [inline lib/features/estimation/presentation/pages/cost_estimation_landing_page.dart]: In my opinion this is completely different task(out of scope), so let's use our hard coded id for now and put this todo link https://ripplearc.youtrack.cloud/issue/CA-467/Refactor-Cost-Estimation-Landing-Page-to-retrieve-Project-ID-via-Bloc.

File: lib/app/shell/app_shell_page.dart [hunk 17]
```diff
@@ -111,44 +104,45 @@ class _AppShellPageState extends State<AppShellPage> {
...
-        ),
-      );
-    }
-
-    return Modular.get<ProjectUIProvider>().buildProjectHeaderAppbar(
-      projectId: projectId,
+    return PreferredSize(
+      preferredSize: const Size.fromHeight(kToolbarHeight),
+      child: StreamBuilder<String?>(
+        stream: _currentProjectNotifier.onCurrentProjectChanged,
```
Comment (SimonDerejeW) [inline lib/app/shell/app_shell_page.dart]: I don't think we should use the notifier here! Since it's out of your scope, I think for now we can put a hard coded string id temporarily, and put this TODO link https://ripplearc.youtrack.cloud/issue/CA-621 to remind us to refactor it.

File: test/app/shell/app_shell_page_test.dart [hunk 18]
```diff
@@ -164,13 +168,13 @@ void main() {
       await tester.pump();
       await tester.pump(const Duration(seconds: 1));
```
Comment (SimonDerejeW) [inline test/app/shell/app_shell_page_test.dart]: I think using `pumpAndSettle` is cleaner. So let's use that when ever it's possible.

Comment (SimonDerejeW) [reply test/app/shell/app_shell_page_test.dart]: This applies to other places in this file as well.

**Review by SimonDerejeW — CHANGES_REQUESTED** [review]

File: lib/app/shell/app_shell_page.dart (hunk 17 above)
Comment (ayanasamuel8) [reply lib/app/shell/app_shell_page.dart]: :+1:

File: lib/features/estimation/presentation/pages/cost_estimation_landing_page.dart (hunk 16 above)
Comment (ayanasamuel8) [reply lib/features/estimation/presentation/pages/cost_estimation_landing_page.dart]: :+1:

File: lib/features/estimation/presentation/pages/cost_estimation_landing_page.dart (hunk 15 above)
Comment (ayanasamuel8) [reply lib/features/estimation/presentation/pages/cost_estimation_landing_page.dart]: :+1:

File: test/app/shell/app_shell_page_test.dart (hunk 18 above)
Comment (ayanasamuel8) [reply test/app/shell/app_shell_page_test.dart]: :+1:

File: lib/app/shell/app_shell_page.dart [hunk 19]
```diff
@@ -0,0 +1,198 @@
...
+  void initState() {
+    super.initState();
+
+    _moduleLoader.ensureTabModuleLoaded(ShellTab.home).then((_) {
+      if (!mounted) return;
+
+      final dropdownBloc = Modular.get<ProjectDropdownBloc>();
+      dropdownBloc.add(const ProjectDropdownStarted());
+
+      // TODO: https://ripplearc.youtrack.cloud/issue/CA-621
```
Comment (ripplearcgit) [inline lib/app/shell/app_shell_page.dart]: 🔴 **Blocking — TODO format wrong (two occurrences in this file).**

Both TODOs in this file (`initState` and `_buildAppBar`) are missing the `[CA-XXX]` tag and description text. Required format:

```dart
// TODO: [CA-621] Wire ProjectDropdownBloc result into app bar via CurrentProjectNotifier.
// https://ripplearc.youtrack.cloud/issue/CA-621
```

Bare URL-only TODOs are not searchable by ticket and do not satisfy the project deferred-work rule.

File: lib/features/estimation/estimation_module.dart [hunk 20]
```diff
@@ -15,34 +15,27 @@ import 'package:construculator/features/estimation/presentation/bloc/cost_estima
...
 import 'package:construculator/features/estimation/presentation/bloc/rename_estimation_bloc/rename_estimation_bloc.dart';
-import 'package:construculator/features/estimation/presentation/pages/cost_estimation_details_page.dart';
 import 'package:construculator/features/estimation/presentation/pages/cost_estimation_landing_page.dart';
-import 'package:construculator/libraries/router/guards/auth_guard.dart';
-import 'package:construculator/libraries/router/routes/estimation_routes.dart';
+import 'package:construculator/libraries/auth/auth_library_module.dart';
+import 'package:construculator/libraries/time/clock_module.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:flutter_modular/flutter_modular.dart';
```
Comment (ripplearcgit) [inline lib/features/estimation/estimation_module.dart:20]: 🔴 **Blocking — TODO format wrong.**

Required format:

```dart
// TODO: [CA-467] Refactor landingPage() to retrieve projectId via Bloc instead of hardcoding.
// https://ripplearc.youtrack.cloud/issue/CA-467
```

File: lib/features/estimation/presentation/pages/cost_estimation_landing_page.dart [hunk 21]
```diff
@@ -21,6 +21,7 @@ import 'package:flutter_modular/flutter_modular.dart';
 import 'package:ripplearc_coreui/ripplearc_coreui.dart';
 
 class CostEstimationLandingPage extends StatefulWidget {
+  // TODO: https://ripplearc.youtrack.cloud/issue/CA-467/Refactor-Cost-Estimation-Landing-Page-to-retrieve-Project-ID-via-Bloc
```
Comment (ripplearcgit) [inline lib/features/estimation/presentation/pages/cost_estimation_landing_page.dart]: 🔴 **Blocking — TODO format wrong.**

Required format:

```dart
// TODO: [CA-467] Remove projectId constructor parameter once retrieved via Bloc.
// https://ripplearc.youtrack.cloud/issue/CA-467
```

File: lib/app/shell/app_shell_page.dart [hunk 22]
```diff
@@ -0,0 +1,198 @@
...
+    if (state.selectedTabIndex != 0) {
+      _bloc.add(const AppShellTabSelected(0));
+      return;
+    }
+
+    SystemNavigator.pop();
+  }
+
+  Future<void> _handleTabTap(int index) async {
+    final tab = ShellTab.values[index];
```
Comment (ripplearcgit) [inline lib/app/shell/app_shell_page.dart]: 🔴 **Blocking — `ensureTabModuleLoaded()` is orchestration logic that belongs in `AppShellBloc`, not the Page.**

`_handleTabTap` currently calls `_moduleLoader.ensureTabModuleLoaded(tab)` directly before dispatching the BLoC event. This means the Page owns the decision of *when* a module gets loaded — that is side-effectful business logic, not presentation.

`TabModuleManager` should be injected into `AppShellBloc` and called inside `_onTabSelected`:

```dart
// ShellModule.binds():
i.addSingleton<TabModuleManager>(() => TabModuleManager(appBootstrap));
i.add<AppShellBloc>(() => AppShellBloc(moduleLoader: i.get()));

// AppShellBloc:
class AppShellBloc extends Bloc<AppShellEvent, AppShellState> {
  final TabModuleManager _moduleLoader;

  AppShellBloc({required TabModuleManager moduleLoader})
      : _moduleLoader = moduleLoader,
        super(const AppShellState(selectedTabIndex: 0, loadedTabIndexes: {0})) {
    on<AppShellTabSelected>(_onTabSelected);
  }

  Future<void> _onTabSelected(
    AppShellTabSelected event,
    Emitter<AppShellState> emit,
  ) async {
    if (event.index == state.selectedTabIndex) return;
    final tab = ShellTab.values[event.index];
    await _moduleLoader.ensureTabModuleLoaded(tab);
    emit(state.copyWith(
      selectedTabIndex: event.index,
      loadedTabIndexes: {...state.loadedTabIndexes, event.index},
    ));
  }
}

// Page — _handleTabTap becomes trivial:
void _handleTabTap(int index) {
  _bloc.add(AppShellTabSelected(index));
}
```

With this change the Page no longer needs `_moduleLoader` as a field, and the `TabModuleManager` import is removed from the Page entirely.

**Review by ripplearcgit — CHANGES_REQUESTED** [review]
**Round 6 — Changes Requested**

Four blockers this round: three TODO annotation format violations and one architectural violation where module-loading orchestration lives in the Page instead of the BLoC. The non-blocking findings are posted separately.

**Review by ripplearcgit — COMMENTED** [review]
🟡 **Non-blocking findings**

**1. Missing `///` doc on `TabModuleManager` (`lib/app/shell/tab_module_manager.dart`)**

The class-level doc was removed in a prior iteration. Please restore:

```dart
/// Manages lazy loading of feature modules for each shell tab.
///
/// Modules are provided via [TabModuleProvider]. Tests can supply lightweight
/// fake providers; production uses [_ProductionTabModuleProvider] by default.
class TabModuleManager { ... }
```

---

**2. `ProjectDropdownBloc` is a Dashboard-feature concern leaking into the shell layer (`lib/app/shell/app_shell_page.dart`)**

`AppShellPage` imports `features/dashboard/presentation/bloc/project_dropdown_bloc/project_dropdown_bloc.dart` and dispatches `ProjectDropdownStarted` from `initState`. This is a cross-layer violation — the app shell should not know about a feature-layer BLoC.

Correct fix (can land in CA-621): move the `ProjectDropdownStarted` dispatch into `DashboardModule.binds()` as an eager singleton, so the BLoC starts itself when the Home tab module is loaded. `AppShellPage` then drops the `ProjectDropdownBloc` import, the `_dropdownSubscription` field, and the `dart:async` import entirely.

Longer-term, `ProjectDropdownBloc` belongs in a dedicated `AppHeaderModule` that the shell imports, with its result wired to `CurrentProjectNotifier` internally — the right permanent home now that `AppShellPage` owns the app bar.

File: lib/app/shell/app_shell_page.dart (hunk 19 above)
Comment (ayanasamuel8) [reply lib/app/shell/app_shell_page.dart]: :+1:

File: lib/features/estimation/estimation_module.dart (hunk 20 above)
Comment (ayanasamuel8) [reply lib/features/estimation/estimation_module.dart:20]: :+1:

File: lib/features/estimation/presentation/pages/cost_estimation_landing_page.dart (hunk 21 above)
Comment (ayanasamuel8) [reply lib/features/estimation/presentation/pages/cost_estimation_landing_page.dart]: :+1:

File: lib/app/shell/app_shell_page.dart (hunk 22 above)
Comment (ayanasamuel8) [reply lib/app/shell/app_shell_page.dart]: :+1:

**Review by ripplearcgit — APPROVED** [review]
**Round 7 — LGTM ✅**

All Round 6 blockers resolved cleanly:

- TODO annotations for CA-621 and CA-467 are correctly formatted with ticket tag, description, and URL.
- `ensureTabModuleLoaded` fully moved into `AppShellBloc._onTabSelected` — `TabModuleManager` injected via constructor, registered in `ShellModule.binds()`. Page is now genuinely presentation-only: only `_bloc` and `_tabNavigatorKeys` remain as fields.
- `ProjectDropdownBloc` import, `_dropdownSubscription`, and `dart:async` all gone from the Page.
- `TabModuleManager` class-level `///` doc restored.
- `ShellTab` enum correctly relocated to `module_model.dart` with re-export for compatibility.
- BLoC test updated correctly with `NoOpTabModuleProvider` via `_noOpLoader()` — no real module construction in unit tests.

Architecture is now in the right shape: shell contract honoured, BLoC owns orchestration, Page owns presentation, deferred work properly annotated. Ready to merge once the base chain lands.

**Review by SimonDerejeW — APPROVED** [review]

**Comment by ayanasamuel8** [comment]
#runcheck
